app/controllers/batches_controller.rb

```diff
@@ -290,25 +290,24 @@ def print_barcodes
     end
   end
 
+  # Handles printing of the worksheet
   def print # rubocop:todo Metrics/AbcSize
-    @task = Task.find(params[:task_id]) if params[:task_id]
-    @workflow = @batch.workflow
+    @task = Task.find_by(id: params[:task_id])
     @pipeline = @batch.pipeline
     @comments = @batch.comments
+    template = @pipeline.batch_worksheet
 
-    robot_id = params.fetch(:robot_id, @batch.robot_id)
-
-    @robot = robot_id ? Robot.find(robot_id) : Robot.with_verification_behaviour.first
-
-    # Fall-back
-    @robot ||= Robot.first
-
-    if @pipeline.is_a?(CherrypickingPipeline)
+    if template == 'cherrypick_worksheet'
```

Comment:
What's the reason for checking the template rather than the pipeline being a 'CherrypickingPipeline' ? Presume it's OK that `@robot` is only set if it's a cherrypicking one?

Reply:
It was mostly because the instance variables being set here are to be used in the worksheets. @robot only gets used in  cherrypick_worksheet, so we don't need to waste time looking it up for other worksheets. I switched to checking the worksheet explicitly, because it seemed to better cover 'why' we were setting the instance (because its used in the template) whereas the fact its a cherrypicking pipeline just happens to correlate (albeit in a pretty tight way).

I think the 'correct' solution, when you have multiple templates rendered by the same endpoint, is probably to use a separate presenter for each one.